### PR TITLE
Implement setAttributesOfItemAtPath for NSFilePosixPermissions attribute

### PR DESCRIPTION
**Pull request by drewcrawford**
Add test coverage

### CODE REVIEW

File: TestFoundation/TestNSFileManager.swift [hunk 1]
```diff
@@ -137,6 +138,26 @@ class TestNSFileManger : XCTestCase {
         }
     }
     
+    func test_setFileAttributes() {
+        let path = "/tmp/testfile"
```
Comment (phausler) [inline TestFoundation/TestNSFileManager.swift]: we might not want to use this since it would potentially conflict with other tests running on CI

Comment (drewcrawford) [inline TestFoundation/TestNSFileManager.swift]: I stole this from the `get` test that was already there.

Can you recommend another design, and I can fix both of them?

Comment (phausler) [inline TestFoundation/TestNSFileManager.swift]: using some sort of unique identifier per-run would be nice to have as a common accessor pattern. perhaps `/tmp/<some name identifying the test><uuid>` might be good as a pattern

**Comment by phausler** [comment]
@swift-ci Please test

**Pull request merged**